### PR TITLE
REALMC-7165 Add secrets delete command to realm-cli 

### PR DESCRIPTION
**Pull request by jo-tse-mongo**
[WIP] in regards to tests

### CODE REVIEW

File: internal/commands/secrets/delete.go
Comment (makesitgo) [inline internal/commands/secrets/delete.go:80]: can we change this function signature to be:

```go
func (i *deleteInputs) resolveSecrets(ui terminal.UI, client realm.Client, app realm.App) ([]realm.Secret, error) {
  // fetch the appSecrets via the realm.Client, you probably want these put into a map[name]Secret

  // if len(i.Secrets) == 0, then we'll want to prompt the user to select (many) from the list of app secrets

  // else iterate over each i.Secret flag and figure out its corresponding ID/realm.Secret

  // return that resolved array of realm.Secret back to the command handler
}
```

basically just shifting around the bits of logic you already have.  and we shouldn't be using `args` for anything here, just rely on the flags setting values

---

also this should move to the `delete_inputs.go` file

File: internal/commands/secrets/delete.go
Comment (makesitgo) [inline internal/commands/secrets/delete.go:133]: I'm sure tests will prove this out, but let's make sure having `nil` values for `details`/`deleted` is ok when the table renders as output text.  Check with @kpatel71716 and @nathantfrank on where we landed with rendering `bool` values in the table (was it as "yes"/"no" or something else?), and then make sure we're consistent here

File: internal/commands/secrets/delete.go
Comment (makesitgo) [inline internal/commands/secrets/delete.go:27]: this should move to the `delete_inputs.go` file

File: internal/commands/secrets/output.go
Comment (makesitgo) [inline internal/commands/secrets/output.go:32]: check out the latest code in the `user` package with how they're handling `headers` and `rows`.  I think it super close to the way you have it here, but can we adjust this approach so it can be similarly used in `secrets list`?

**Review by makesitgo — COMMENTED** [review]
just a quick pass here... just a few small things for now.  lookin real good so far!

File: internal/commands/secrets/delete.go
Comment (jo-tse-mongo) [reply internal/commands/secrets/delete.go:133]: From what I remember about `parseValue` is that it can take in nil, but it'll leave it as an empty string

Comment (makesitgo) [reply internal/commands/secrets/delete.go:133]: i think i wrote this comment on the wrong line, but I was more concerned with the `bool` vs `nil` treatment.  I think we want `true`/`false` to render as `"yes"`/`"no"`?

File: internal/cloud/realm/secrets.go
Comment (makesitgo) [inline internal/cloud/realm/secrets.go:33]: we definitely still want this here, as we read from the `res.Body` next here

File: internal/cloud/realm/secrets.go
Comment (makesitgo) [inline internal/cloud/realm/secrets.go:81]: and you won't need the `defer` here, since we never actually attempt to read the `res.Body` anywhere here

File: internal/cloud/realm/secrets_test.go
Comment (makesitgo) [inline internal/cloud/realm/secrets_test.go:54]: we probably don't even need all of this here (though it's great testing!).  for the most part, leave the file as is, and just add a few extra tests... it should read something like:

```go
t.Run("should fail without an auth client", func(t *testing.T) {
  
  t.Run("with an active session ", func(t *testing.T) {

    // FIXME: actually we can just delete this test case now, its effectively duplicated with the new delete stuff you'll be adding
    t.Run("should have no secrets upon app initialization", func(t *testing.T) {

    t.Run("should create a secret", func(t *testing.T) {
      t.Run("and list all app secrets", func(t *testing.T) {})
    })

    t.Run("should delete a secret", func(t *testing.T) {

      // TODO: delete the created test secret here

      t.Run("and list all app secrets", func(t *testing.T) {

        // TODO: find all test secrets again and assert it's empty

      })
    })
    })
  })
})
```

File: internal/commands/secrets/delete.go
Comment (makesitgo) [inline internal/commands/secrets/delete.go:49]: NOTE: once @kpatel71716 merges his `user revoke` work, we'll have an established pattern for commands structured like this going forward.  i think it'll be worth your time here to make the necessary refactors to line up with that pattern, so going forward `users` AND `secrets` can be considered "scope-complete"

File: internal/commands/secrets/output.go
Comment (makesitgo) [inline internal/commands/secrets/output.go:14]: is this used anywhere?

File: internal/commands/secrets/output.go
Comment (makesitgo) [inline internal/commands/secrets/output.go:33]: [question] could this just be `func secretHeaders(additionalHeaders ...string) []string`?

File: internal/commands/secrets/delete_inputs.go
Comment (makesitgo) [inline internal/commands/secrets/delete_inputs.go:32]: can we flip this structure a bit for readability purposes? something like:

```
func (i *deleteInputs) resolveSecrets(ui terminal.UI, secrets []realm.Secret) ([]realm.Secret, error) {
  if len(i.secrets) > 0 {
    resolved := make([]realm.Secret, len(i.secrets))
    // ...
    return resolved, nil
  }

  // no "else" needed here since we return above, less indentation for the next code path
}
```

File: internal/commands/secrets/delete_inputs.go
Comment (makesitgo) [inline internal/commands/secrets/delete_inputs.go:49]: [stylenit] omit this newline, also optionally rename this to `err` and/or inline the if-call

File: internal/commands/secrets/delete_inputs.go
Comment (makesitgo) [inline internal/commands/secrets/delete_inputs.go:73]: the first arg (`_`_ here) is actually the value itself, so you can use that instead of `names[arg]` a 2nd time here (same for `id below)

File: internal/commands/secrets/delete_inputs_test.go
Comment (makesitgo) [inline internal/commands/secrets/delete_inputs_test.go:43]: a note about these `t.Run` strings: try to keep things lowercase and avoid using punctation like `.` or `:` in  (it ends up looking a bit wonky in the resulting `go test` output)

something like `"should return the secrets when specified by"+tc.description` would suffice here

File: internal/commands/secrets/delete_inputs_test.go
Comment (makesitgo) [inline internal/commands/secrets/delete_inputs_test.go:81]: really neat! 🔥

File: internal/commands/secrets/delete_test.go
Comment (makesitgo) [inline internal/commands/secrets/delete_test.go:44]: [stylenit] these are small enough to be defined inline: `{secrets[1], nil}` (and this particular `[]secretOutput` array can probably fit all in-line

File: internal/commands/secrets/delete_test.go
Comment (makesitgo) [inline internal/commands/secrets/delete_test.go:120]: [question] are these used anywhere else? they probably can just be defined right in `clientSetup`

File: internal/commands/secrets/delete_test.go
Comment (makesitgo) [inline internal/commands/secrets/delete_test.go:156]: [stylenit] could make this all one line

**Review by makesitgo — COMMENTED** [review]
bravo!  going to comment here and wait for the last few changes here, but nothing blocking by me.  this looks great!

File: internal/cloud/realm/secrets.go
Comment (jo-tse-mongo) [reply internal/cloud/realm/secrets.go:33]: oops this was a mistake

File: internal/commands/secrets/delete_test.go
Comment (jo-tse-mongo) [reply internal/commands/secrets/delete_test.go:120]: they're used multiple times in the different test cases for this set

File: internal/commands/secrets/output.go
Comment (jo-tse-mongo) [reply internal/commands/secrets/output.go:14]: The table displays some message before the actual table itself, so it's used there
It's called "message" so I'm not sure what to name it or if it should be a const

Comment (makesitgo) [reply internal/commands/secrets/output.go:14]: we can remove the `const` and just leave the string value as an arg to `terminal.NewTable(...`

File: internal/commands/commands.go
Comment (kpatel71716) [inline internal/commands/commands.go:154]: [nit] We should probably use `secrets delete` here to be consistent with pluralization

File: internal/cloud/realm/secrets.go
Comment (kpatel71716) [inline internal/cloud/realm/secrets.go:72]: [nit/reminder] that we are going with just `err` here for naming standards

**Review by kpatel71716 — COMMENTED** [review]
Looks pretty good to me! Just some really small things

File: internal/cloud/realm/secrets_test.go
Comment (makesitgo) [inline internal/cloud/realm/secrets_test.go:51]: [optional/style] personally I'd just nest this further under `should create a secret` and avoid the `var secret realm.Secret` declaration at the top

**Review by makesitgo — APPROVED** [review]
ibueno!

🆒  beans

**Comment by makesitgo** [comment]
we will wait for @kpatel71716 to merge in #54 so you can rebase off his test fixes, and then evergreen should be good to go for this

**Pull request merged**
